lib/query.js

```diff
@@ -3922,6 +3931,10 @@ Query.prototype._findAndModify = function(type, callback) {
     if (err) {
       return callback(err);
     }
+    const mongooseOptions = _this._mongooseOptions;
```

Comment:
Move this logic into `_completeOne()` as well since it is the same as the logic my other comment refers to